relay-redis/src/real.rs

```diff
@@ -1,146 +1,209 @@
+    ///
+    /// This method initializes a connection client that can communicate with multiple Redis nodes
+    /// in a cluster configuration. The client is configured with the specified servers and options.
+    ///
+    /// The client uses a custom cluster manager that implements a specific connection recycling
+    /// strategy, ensuring optimal performance and reliability in cluster environments.
+    pub fn cluster<'a>(
         servers: impl IntoIterator<Item = &'a str>,
         opts: &RedisConfigOptions,
     ) -> Result<Self, RedisError> {
-        AsyncRedisConnection::cluster(servers, opts)
-            .await
-            .map(AsyncRedisClient::new)
+        let servers = servers
+            .into_iter()
+            .map(|s| s.to_string())
+            .collect::<Vec<_>>();
+
+        // We use our custom cluster manager which performs recycling in a different way from the
+        // default manager.
+        let manager = pool::CustomClusterManager::new(servers, false, opts.refresh_interval)
+            .map_err(RedisError::Redis)?;
+
+        let pool = Self::build_pool(manager, opts)?;
+
+        Ok(AsyncRedisClient::Cluster(pool))
     }
 
-    /// Creates a new [`AsyncRedisClient`] in single mode.
-    pub async fn single(server: &str, opts: &RedisConfigOptions) -> Result<Self, RedisError> {
-        AsyncRedisConnection::single(server, opts)
-            .await
-            .map(AsyncRedisClient::new)
+    /// Creates a new connection client for a single Redis instance.
+    ///
+    /// This method initializes a connection client that communicates with a single Redis server.
+    /// The client is configured with the specified server URL and options.
+    ///
+    /// The client uses a custom single manager that implements a specific connection recycling
+    /// strategy, ensuring optimal performance and reliability in single-instance environments.
+    pub fn single(server: &str, opts: &RedisConfigOptions) -> Result<Self, RedisError> {
+        // We use our custom single manager which performs recycling in a different way from the
+        // default manager.
+        let manager = pool::CustomSingleManager::new(server, opts.refresh_interval)
+            .map_err(RedisError::Redis)?;
+
+        let pool = Self::build_pool(manager, opts)?;
+
+        Ok(AsyncRedisClient::Single(pool))
     }
 
-    /// Returns a shared [`AsyncRedisConnection`].
-    pub fn get_connection(&self) -> AsyncRedisConnection {
-        self.connection.clone()
+    /// Acquires a connection from the pool.
+    ///
+    /// Returns a new [`AsyncRedisConnection`] that can be used to execute Redis commands.
+    /// The connection is automatically returned to the pool when dropped.
+    pub async fn get_connection(&self) -> Result<AsyncRedisConnection, RedisError> {
+        let connection = match self {
+            Self::Cluster(pool) => {
+                AsyncRedisConnection::Cluster(pool.get().await.map_err(RedisError::Pool)?)
```

Comment:
Does this call need to be `timeout_get` instead of `get`?

Reply:
It uses `timeout_get` inside.